packages/ia-topnav/src/lib/keyboard-navigation.js

```diff
@@ -0,0 +1,128 @@
+
+      const normalElements = Array.from(waybackSlider.querySelectorAll(focusableTagSelectors));
+
+      // wayback save-form focusable elements
+      const savePageForm = waybackSlider.querySelector('save-page-form');
+      const savePageFormElements = Array.from(savePageForm.shadowRoot.querySelectorAll(focusableTagSelectors));
+
+      elements = [...waybackSearchElements, ...normalElements, ...savePageFormElements];
+    } else {
+      elements = this.elementsContainer.querySelectorAll(focusableTagSelectors);
+    }
+
+    return Array.from(elements).filter(isDisabledOrHidden);
+  }
+
+  /**
+   * Handles keyboard events and focuses the appropriate element.
+   * @param {KeyboardEvent} event - The keyboard event object.
+   */
+  handleKeyDown(event) {
+    const { key } = event;
+    const isArrowKey = ['ArrowDown', 'ArrowRight', 'ArrowUp', 'ArrowLeft'].includes(key);
+    const isTabKey = key === 'Tab';
+
+    if (isArrowKey) {
+      this.handleArrowKey(key);
+      event.preventDefault();
+    } else if (isTabKey) {
+      this.handleTabKey(event);
+    }
+  }
+
+  /**
+   * Handles arrow key events and focuses the next or previous element.
+   * @param {string} key - The key that was pressed ('ArrowDown', 'ArrowRight', 'ArrowUp', or 'ArrowLeft').
+   */
+  handleArrowKey(key) {
+    const isDownOrRight = ['ArrowDown', 'ArrowRight'].includes(key);
+    isDownOrRight ? this.focusNext() : this.focusPrevious();
+  }
+
+  /**
+   * Focuses the previous focusable element in the container.
+   */
+  focusPrevious() {
+    if (this.focusableElements.length === 0) return;
+    this.focusedIndex = (this.focusedIndex - 1 + this.focusableElements.length) % this.focusableElements.length;
+    this.focusableElements[this.focusedIndex]?.focus();
+  }
+
+  /**
+   * Focuses the next focusable element in the container.
+   */
+  focusNext() {
+    if (this.focusableElements.length === 0) return;
+    this.focusedIndex = (this.focusedIndex + 1) % this.focusableElements.length;
+    this.focusableElements[this.focusedIndex]?.focus();
+  }
+
+  /**
+   * Handles the Tab key event and focuses the next or previous menu item.
+   * @param {KeyboardEvent} event - The keyboard event object.
+   */
+  handleTabKey(event) {
+    if (this.menuOption) {
+      const isShiftPressed = event.shiftKey;
+      this.focusNextMenuItem(isShiftPressed);
+    }
+
+    this.focusableElements[this.focusedIndex]?.blur();
+    event.preventDefault();
+  }
+
+  /**
+   * Focuses the next or previous menu item based on the provided flag.
+   * @param {boolean} isPrevious - A flag indicating whether to focus the previous menu item.
+   */
+  focusNextMenuItem(isPrevious = false) {
+    this.elementsContainer.dispatchEvent(
+      new CustomEvent('focusToNext', {
+        bubbles: true,
+        composed: true,
+        detail: {
+          mediatype: this.menuOption,
+          moveTo: isPrevious ? 'prev' : 'next',
+        },
```

Comment:
I find the naming a bit inconsistent here, since above we have separate `focusNext` and `focusPrevious` methods, yet here we have only a single `focusNextMenuItem` method which handles both directions. I'd argue that consistency & clarity should take precedence over conciseness here, though perhaps only a minor concern.

Reply:
Make sense, I am changing function name. BTW, this function is to move focus on other parent items, where the above function moves focus in within a container like subnav or usermenu element